### ISSUE TITLE
Strings derived from `STDOUT` have unexpected `\n` character in Windows

### ISSUE THREAD

**Issue by MadhukaHarith92** [ballerina-platform/ballerina-standard-library#1476]
**Description:**
$Subject

**Steps to reproduce:**
Enable observability and run the below sample.
```
import ballerina/io;
import ballerina/observe;
import ballerina/observe.mockextension;

public function main() {
    var spans = mockextension:getFinishedSpans("Ballerina");
    io:println(spans[0].traceId);
    io:println(spans[0].spanId);
}
```
**Config.toml**:
```
[ballerina.observe]
enabled=true
provider="mock"
```

Read the `STDOUT` stream and assign it to a variable `outStream`. Then run below sample.
```
import ballerina/io;
import ballerina/regex;

public function main() {
    string[] lines = regex:split(outStream, "\n");
    string traceId = lines[1];
    string spanId = lines[2];
    io:println("traceId: " + traceId + "spanId: " + spanId);
}
``` 

**Actual output:**
```
traceId: 4c2a67c8568f1d3d56b68897fa56ba84
 spanId: c72b6990c84111c8
```

**Expected output:**
```
traceId: 4c2a67c8568f1d3d56b68897fa56ba84 spanId: c72b6990c84111c8
```

The output is as expected in `ubuntu`.
**Affected Versions:**
SL Beta1

**OS, DB, other environment details and versions:**
Windows

**Comment by BuddhiWathsala** [comment]
Tried the following example code block in Windows to reproduce the abovementioned issue. But the mentioned issue did not appear to me.
```ballerina
string id1 = "A001";
string id2 = "A002";
io:println("ID 1: " + id1 + "ID 1: " + id2);
```
@MadhukaHarith92, is there any proper code block that can be used to reproduce this issue?.

**Comment by MadhukaHarith92** [comment]
> Tried the following example code block in Windows to reproduce the abovementioned issue. But the mentioned issue did not appear to me.
> 
> ```ballerina
> string id1 = "A001";
> string id2 = "A002";
> io:println("ID 1: " + id1 + "ID 1: " + id2);
> ```
> 
> @MadhukaHarith92, is there any proper code block that can be used to reproduce this issue?.

1. Get a checkout from https://github.com/ballerina-platform/module-ballerina-log/tree/choreo-tracing branch.
2. Modify  https://github.com/ballerina-platform/module-ballerina-log/blob/choreo-tracing/log-integration-tests/tests/resources/samples/observability-project-logfmt/main.bal as follows.

Replace
`io:print("traceId = \"" + spans[0].traceId + "\" spanId = \"" + spans[0].spanId + "\"");` 

with 
```
io:println(spans[0].traceId);
io:println(spans[0].spanId);
```
3. Print the `traceId` and `spanId` in https://github.com/ballerina-platform/module-ballerina-log/blob/choreo-tracing/log-integration-tests/tests/tests_logfmt.bal#L306 together using println.

**Comment by BuddhiWathsala** [comment]
I was able to reproduce this issue. To solve this, replace `characterChannel.read(1000)` from `characterChannel.readString()` as [this in commit](https://github.com/ballerina-platform/module-ballerina-log/commit/8e9f71aef91cf3b7549591c9ada50927e586d77c).

The reason for this issue is incompatible line termination in windows and Linux.
- windows line terminator is `\r\n` 
- linux line terminator is `\n`

Even though you split using `\n`, the `\r` character can be present. That is the cause. In the `readString` API, we have enabled a normalization step which replace all the platform specific line terminations with newline character(`\n`) as per this issue https://github.com/ballerina-platform/ballerina-standard-library/issues/1392. However, we didn't change it in read API because someone might need to read the entire character `\r\n` without alteration.
